packages/flutter_tools/test/general.shard/build_system/targets/dart_plugin_registrant_test.dart

```diff
@@ -0,0 +1,318 @@
+
+dependencies:
+  flutter:
+    sdk: flutter
+  path_provider_linux: 1.0.0
+''';
+
+const String _kLinuxRegistrant =
+'//\n'
+'// Generated file. Do not edit.\n'
+'//\n'
+'\n'
+'// @dart = 2.12\n'
+'\n'
+'import \'package:path_provider_example/main.dart\' as entrypoint;\n'
+'import \'dart:io\'; // flutter_ignore: dart_io_import.\n'
+'import \'package:path_provider_linux/path_provider_linux.dart\';\n'
+'\n'
+'@pragma(\'vm:entry-point\')\n'
+'class _PluginRegistrant {\n'
+'\n'
+'  @pragma(\'vm:entry-point\')\n'
+'  static void register() {\n'
+'    if (Platform.isLinux) {\n'
+'      try {\n'
+'        PathProviderLinux.registerWith();\n'
+'      } catch (err) {\n'
+'        print(\n'
+'          \'`path_provider_linux` threw an error: \$err. \'\n'
+'          \'The app may not function as expected until you remove this plugin from pubspec.yaml\'\n'
+'        );\n'
+'        rethrow;\n'
+'      }\n'
+'\n'
+'    } else if (Platform.isMacOS) {\n'
+'    } else if (Platform.isWindows) {\n'
+'    }\n'
+'  }\n'
+'\n'
+'}\n'
+'\n'
+'typedef _UnaryFunction = dynamic Function(List<String> args);\n'
+'typedef _NullaryFunction = dynamic Function();\n'
+'\n'
+'void main(List<String> args) {\n'
+'  if (entrypoint.main is _UnaryFunction) {\n'
+'    (entrypoint.main as _UnaryFunction)(args);\n'
+'  } else {\n'
+'    (entrypoint.main as _NullaryFunction)();\n'
+'  }\n'
+'}\n'
+'';
+
+const String _kSamplePluginPubspec = '''
+name: path_provider_linux
+description: linux implementation of the path_provider plugin
+// version: 2.0.1
+// homepage: https://github.com/flutter/plugins/tree/master/packages/path_provider/path_provider_linux
+
+flutter:
+  plugin:
+    implements: path_provider
+    platforms:
+      linux:
+        dartPluginClass: PathProviderLinux
+        pluginClass: none
+
+environment:
+  sdk: ">=2.12.0-259.9.beta <3.0.0"
+  flutter: ">=1.20.0"
+''';
+
+void main() {
+
+  group('Dart plugin registrant' , () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+
+    testWithoutContext('skipped based on environment.generateDartPluginRegistry',
+        () async {
+      final Environment environment = Environment.test(
+          fileSystem.currentDirectory,
+          artifacts: null,
+          fileSystem: fileSystem,
+          logger: BufferLogger.test(),
+          processManager: FakeProcessManager.any(),
+          generateDartPluginRegistry: false);
+
+      expect(const DartPluginRegistrantTarget().canSkip(environment), true);
+
+      final Environment environment2 = Environment.test(
+          fileSystem.currentDirectory,
+          artifacts: null,
+          fileSystem: fileSystem,
+          logger: BufferLogger.test(),
+          processManager: FakeProcessManager.any(),
+          generateDartPluginRegistry: true);
+
+      expect(const DartPluginRegistrantTarget().canSkip(environment2), false);
+    });
+
+    testUsingContext("doesn't generate generated_main.dart if there aren't Dart plugins", () async {
```

Comment:
Here and below, just asking to confirm that these tests can't be `testWithoutContext`.

Reply:
There are a few global dependencies that touches other functions as well. I managed to get these three tests to use `testWithoutContext`, but in doing so, I broke almost all the plugins test.